source/m2mnsdlinterface.cpp

```diff
@@ -557,8 +557,10 @@ uint8_t M2MNsdlInterface::received_from_server_callback(struct nsdl_s *nsdl_hand
                 tr_debug("M2MNsdlInterface::received_from_server_callback - Empty ACK, msg id: %d", coap_header->msg_id);
                 M2MBase *base = find_resource("", coap_header->token_ptr, coap_header->token_len);
                 if (base) {
-                    M2MResource* res = static_cast<M2MResource*> (base);
-                    res->notification_sent();
+                    if (M2MBase::Resource == base->base_type()) {
```

Comment:
If this is only needed on resource, not resource instance, perhaps a comment would be in place?